web/profiles/joinup/joinup.info.yml

```diff
@@ -39,7 +40,6 @@ dependencies:
   - rdf_entity
   - rdf_export
   - rdf_taxonomy
-  - solution
   - state_machine
```

Comment:
No it's fine, I just checked it. The modules are installed in `install_profile_modules()`. This retrieves the full module list from `system_rebuild_module_data` which has the proper dependencies resolved in it. Every module gets assigned a weight (taken from the `->sort` parameter in the module data), and this is used then to sort the profile's module. The actual order of the modules in the dependency list is only used for modules that are on the same level (i.e. that have the same dependencies).

Here are the weights that are given to the modules you mention during the actual install:

* `collection`: -45
* `joinup_news`: -117
* `og`: -40
* `solution`: -112

So according to this list, `og` will be installed first, then `collection` -> `solution` -> `joinup_news`